sql/core/src/main/java/org/apache/spark/sql/execution/datasources/parquet/VectorizedColumnReader.java

```diff
@@ -145,7 +146,24 @@ public VectorizedColumnReader(
     DictionaryPage dictionaryPage = pageReader.readDictionaryPage();
     if (dictionaryPage != null) {
       try {
-        this.dictionary = dictionaryPage.getEncoding().initDictionary(descriptor, dictionaryPage);
+        PrimitiveType primitiveType = descriptor.getPrimitiveType();
+        if (primitiveType.getOriginalType() == OriginalType.DECIMAL &&
+            primitiveType.getDecimalMetadata().getPrecision() <= Decimal.MAX_INT_DIGITS() &&
+            primitiveType.getPrimitiveTypeName() == PrimitiveType.PrimitiveTypeName.INT64) {
```

Comment:
nit: we can directly use `INT64` as it's imported.

Reply:
we will have to add a static import because it's not imported at the moment. I can do that for code-clarity

The reason why it's working in other places in this file without the qualified name is because Java requires case statements to have `UNqualified` names of enums